src/lib/operators/import_binary.cpp

```diff
@@ -117,6 +144,51 @@ std::pair<std::shared_ptr<Table>, ChunkID> ImportBinary::_read_header(std::ifstr
   return std::make_pair(table, chunk_count);
 }
 
+std::shared_ptr<AbstractPartitionSchema> ImportBinary::_read_partitioning_header(std::ifstream& file) {
+  const auto partition_schema_type = partition_schema_type_to_uint.right.at(_read_value<uint8_t>(file));
+  const auto partition_count = _read_value<PartitionID>(file);
+
+  switch (partition_schema_type) {
+    case PartitionSchemaType::Null: {
+      return std::make_shared<NullPartitionSchema>();
+    }
+    case PartitionSchemaType::RoundRobin: {
+      return std::make_shared<RoundRobinPartitionSchema>(partition_count);
+    }
+    case PartitionSchemaType::Hash: {
+      const auto column_id = _read_value<ColumnID>(file);
+      const HashFunction hash_function;
+      return std::make_shared<HashPartitionSchema>(column_id, hash_function, partition_count);
```

Comment:
I don't get this part. You probably have `HashFunction` so that you can support different hash functions. If you create your own `HashFunction` here no matter what has been exported, won't this break the partitioning?